File: Leetcode/Compare Strings by Frequency of the Smallest Character/Compare Strings by Frequency of the Smallest Character.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> numSmallerByFrequency(vector<string>& queries, vector<string>& words) 
    {
        //declaration of map
         vector<int> maps;
        
        //loop for taking string words into map
        for(int i = 0; i<words.size();i++)
        {
            string word = words[i];
            sort(word.begin(),word.end());
            maps.push_back(count(word.begin(),word.end(),word[0])); 
        }
        
        //sorting map
        sort(maps.begin(),maps.end());
        
        //vector declaration
        vector<int> ans;
        
        //loop to compare strings
        for(int i = 0;i<queries.size();i++)
        {
            string word = queries[i];
            sort(word.begin(),word.end());
            int count_ = count(word.begin(),word.end(),word[0]);
            int upper1 = upper_bound(maps.begin(), maps.end(), count_) - maps.begin(); 
            
            //taking the counted number of words according to the question
            ans.push_back(maps.size()-upper1);
        }
        
        //returning vector
        return ans;
    }
};

File: Leetcode/Compare Strings by Frequency of the Smallest Character/Compare Strings by Frequency of the Smallest Character.py
class Solution(object):
    def numSmallerByFrequency(self, queries, words):
        """
        :type queries: List[str]
        :type words: List[str]
        :rtype: List[int]
        """
        queries_list=[self.calculateFrequency(i) for i in queries]
        words_list=[self.calculateFrequency(i) for i in words]
            
        res=[]
        
        num=0
        while num<len(queries_list):              # Calclating result
            count=0
            for j in words_list:
                if queries_list[num]<j:
                    count+=1
            num+=1
            res.append(count)
            
        return res
            
        
    def calculateFrequency(self,text):            # Sorting the string/text
        text = sorted(text)
        count=collections.Counter(text)

        return count[text[0]]                     # Returning result
